test/framework/src/main/java/org/elasticsearch/index/mapper/FieldTypeTestCase.java

```diff
@@ -283,92 +95,6 @@ public void testEquals() {
         assertEquals(ft1, ft2); // symmetric
         assertEquals(ft2, ft1);
         assertEquals(ft1.hashCode(), ft2.hashCode());
-
-        for (Modifier modifier : modifiers) {
-            ft1 = createNamedDefaultFieldType();
-            ft2 = createNamedDefaultFieldType();
-            modifier.modify(ft2);
-            assertFieldTypeNotEquals(modifier.property, ft1, ft2);
```

Comment:
I just noticed we've lost test coverage for the `equals` method, which is used in `FieldTypeLookup`. This is too bad, I wonder if we could we keep these modifiers around or switch to using `EqualsHashCodeTestUtils#checkEqualsAndHashCode`.

Reply:
With luck we'll be able to simplify a lot of the MapperService lookup stuff once things have been rearranged, but for now I agree we still need these tests.  I've modified FieldTypeTestCase to allow testing of subtypes with modifiers, using `EqualsHashCodeTestUtils`.